test/e2e_node/split_disk_test.go

```diff
@@ -0,0 +1,309 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/kubernetes/pkg/features"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	evictionapi "k8s.io/kubernetes/pkg/kubelet/eviction/api"
+	kubeletmetrics "k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	"k8s.io/kubernetes/test/e2e/nodefeature"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("SplitDisk", nodefeature.SplitDisk, func() {
+	f := framework.NewDefaultFramework("split-disk-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+	pressureTimeout := 10 * time.Minute
+	expectedNodeCondition := v1.NodeDiskPressure
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		e2eskipper.SkipUnlessFeatureGateEnabled(features.KubeletSeparateDiskGC)
+		if !hasSplitFileSystem(ctx) {
+			ginkgo.Skip("it doesn't have split filesystem")
+		}
+	})
+
+	f.It("should display different stats for imagefs and containerfs", func(ctx context.Context) {
+		summary := eventuallyGetSummary(ctx)
+		gomega.Expect(summary.Node.Fs.AvailableBytes).ToNot(gomega.Equal(summary.Node.Runtime.ImageFs.AvailableBytes))
+		gomega.Expect(summary.Node.Fs.CapacityBytes).ToNot(gomega.Equal(summary.Node.Runtime.ImageFs.CapacityBytes))
+		// Node.Fs represents rootfs which /var/lib/kubelet is on.
+		// Since the graphroot is left by default in storage.conf, and it is the same filesystem as rootfs,
+		// Node.Fs should be the same as Runtime.ContainerFs.
+		gomega.Expect(summary.Node.Fs.AvailableBytes).To(gomega.Equal(summary.Node.Runtime.ContainerFs.AvailableBytes))
+		gomega.Expect(summary.Node.Fs.CapacityBytes).To(gomega.Equal(summary.Node.Runtime.ContainerFs.CapacityBytes))
+	})
+
+	f.Context("when there is disk pressure", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), func() {
+		f.Context("on imageFs", func() {
+			tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+				initialConfig.EvictionHard = map[string]string{string(evictionapi.SignalNodeFsAvailable): "30%", string(evictionapi.SignalContainerFsAvailable): "30%", string(evictionapi.SignalImageFsAvailable): "30%"}
```

Comment:
To cut down on the really long line...

```suggestion
initialConfig.EvictionHard = map[string]string{
		string(evictionapi.SignalNodeFsAvailable): "30%",
		string(evictionapi.SignalContainerFsAvailable): "30%",
		string(evictionapi.SignalImageFsAvailable): "30%",
}
```